tests/integration/cli/commands/test_issue_1667.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import os.path
+import subprocess
+import sys
+
+import pytest
+
+from pex.cli.testing import run_pex3
+from pex.testing import run_pex_command
+from pex.typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import Any
+
+
+@pytest.mark.skipif(
+    sys.version_info[:2] < (3, 7) or sys.version_info[:2] >= (3, 11),
+    reason="The lock under test requires an interpreter satisfying >=3.7,<3.11 to test against.",
+)
+def test_interpreter_constraints_range_coverage(tmpdir):
+    # type: (Any) -> None
+
+    # We lock with an unconstrained IPython requirement and we know IPython latest does not support
+    # Python 3.7. If locking respects ICs it should not pick latest, but a version that supports at
+    # least 3.7
+    lock = os.path.join(str(tmpdir), "lock.json")
+    run_pex3(
+        "lock",
+        "create",
+        "--style",
+        "universal",
+        "--resolver-version",
+        "pip-2020-resolver",
+        "--interpreter-constraint",
+        ">=3.7,<3.11",
+        "ipython",
+        "-o",
+        lock,
+    ).assert_success()
+
+    pex_root = os.path.join(str(tmpdir), "pex_root")
+    ipython_pex = os.path.join(str(tmpdir), "ipython.pex")
+    run_pex_command(
+        args=[
+            "--interpreter-constraint",
+            ">=3.7,<3.11",
```

Comment:
If you were to run this test on a machine without Python 3.7 discoverable (such that Pex tries to install for 3.7), I suspect this test would still pass. Note that the offending lockfile from https://github.com/pantsbuild/pex/issues/1667 works when run with Python 3.8+, only not 3.7.

Perhaps one of these strategies instead?
- Force the lockfile consumption to use Python 3.7, or
- inspect the lock's selected version

Reply:
The pip-2020-resolver takes ~infinite time to resolve this under Python 2.7; thus this approach for now. Still looking into why that is the case.

Reply:
IOW, It works great under `tox -epy{py3,{5,6,7,8,9,10}}...`. It stalls, makes incredibly slow progress, but effectively halts, for `tox -epy{py,27}...`.

Reply:
I'm not totally following - setting `==3.7.*` causes Py2 to stall? 

Alternative wording: I'm trying to make sure we can be confident Pex is building with Python 3.7, and not using 3.8+ thanks to the loose constraint of `>=3.7,<3.11`.

Reply:
I'll give up and stick py37 back in on the consume side. It doesn't actually provide any more coverage but answers the sort of question you're having more directly I guess.

Reply:
Fixed.